### ISSUE TITLE
Feature request: Add gf_gen_decode_matrix to the public API

### ISSUE THREAD

**Issue by asomers** [01org/isa-l#27]
ISA-L's public API provides no help to the user for generating a decode matrix for reconstructing missing data.  Practically every single user of the erasure code module will need to do that.  In fact, ISA-L's own test and benchmark programs all do this, copying similar code into 8 separate files.  Could we please have a common helper function for it?  As an example of such a function, see Jerasure's [jerasure_make_decoding_matrix](https://github.com/ceph/jerasure/blob/master/include/jerasure.h#L201).

**Comment by gbtucker** [comment]
Alan, thanks for the feedback.  I've been hesitant to include one generate decode matrix function since its complexity can depend so much on assumptions of how the encode matrix was generated.  Instead I put a few as examples when the encode matrix is symmetric (top portion is identity or source array is un-coded) but even this is not a restriction for other functions.  I'll look into creating something that is generic.

**Comment by zoltan** [comment]
Agreed. Not having this in the API is very confusing for anybody who is looking to start using the library.

Have there been any progress on this?

**Comment by gbtucker** [comment]
We have added some better examples since then.  The `gf_gen_decode_matrix_simple()` in the examples/ec directory is an attempt at keeping this general without too many assumptions on the encoding matrix.  Although it does assume symmetric encoding.
